pkg/plugins/models.go

```diff
@@ -254,6 +256,13 @@ type Error struct {
 	message         string          `json:"-"`
 }
 
+type LoadingStrategy string
+
+const (
+	LoadingStrategyFetch  LoadingStrategy = "fetch"
+	LoadingStrategyScript LoadingStrategy = "script"
```

Comment:
what are fetch and script?  are we aiming to have everything be fetch eventually?

Reply:
@jackw is esm dynamic import distinct from script and fetch/eval?

Reply:
I'm not sure I'm understanding the question @leeoniya . 🤔 Are you asking if we can load plugins via esm dynamic imports?

Reply:
> Are you asking if we can load plugins via esm dynamic imports?

hehe, yes

Reply:
You're not the first to ask this but no we cannot currently load plugins via esm dynamic imports. Doing so would require:

- new tooling to build plugins as esm (as webpack doesn't support it) which would not be backwards compatible with older versions of Grafana (unless we introduce multiple plugin builds).
- not loading via Systemjs as it doesn't support es modules.
- due to not using Systemjs figure out wiring up the shared dependencies via an import-map so esm plugins can access the same state/context/etc.